java/java.hints/test/unit/src/org/netbeans/modules/java/hints/jdk/ConvertToNestedRecordPatternTest.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.java.hints.jdk;
+
+import org.netbeans.junit.NbTestCase;
+import org.netbeans.modules.java.hints.test.api.HintTest;
+import javax.lang.model.SourceVersion;
+
+/**
+ *
+ * @author mjayan
+ */
+public class ConvertToNestedRecordPatternTest extends NbTestCase {
+
+    public ConvertToNestedRecordPatternTest(String name) {
+        super(name);
+    }
+
+    public void testSimple() throws Exception {
+        if (!isRecordClassPresent()) {
+            return;
+        }
+        HintTest.create()
+                .input("package test;\n"
+                        + "record Rect(ColoredPoint upperLeft,ColoredPoint lr) {}\n"
+                        + "record ColoredPoint(Point p, Color c) {}\n"
+                        + "record Point(int x, int y){}\n"
+                        + "enum Color {RED,GREEN,BLUE}\n"
+                        + "public class Test {\n"
+                        + "    private void test(Object o) {\n"
+                        + "        if (o instanceof Rect(ColoredPoint ul, ColoredPoint lr)) {\n"
```

Comment:
One thing to note here is that for JDK 20, the named record patterns will likely be dropped, and only "unnamed" will remain. Might be reasonable to reflect that here even now.

Reply:
this will be handled as part of JDK 20 changes